main-english comments.py

```diff
@@ -1,180 +1,181 @@
-
-    axs[0][0].imshow(original)
-    axs[0][0].set_title("Orginal Image")
-    axs[0][0].axis('off')
-
-    axs[0][1].imshow(convertedImageRow)
-    axs[0][1].set_title("Sorted each Row")
-    axs[0][1].axis('off')
-
-    axs[1][0].imshow(convertedImageColumn)
-    axs[1][0].set_title("Sorted each Column")
-    axs[1][0].axis('off')
-
-    axs[1][1].imshow(convertedImageContinuously)
-    axs[1][1].set_title("Sorted Continuously")
-    axs[1][1].axis('off')
-
-    fig.tight_layout()
-    plt.show()
-
-convert(original)
+from PIL import Image 
+import numpy as np 
+import matplotlib.pyplot as plt 
+
+ 
+### The Quicksort Algorithm 
+# This is a helpermethod for the quickSort Algorithm 
+# We need to give the quickSort the parameters arr, low, high and comparator every time 
+# arr = is the array we want to sort 
+# low = the index of the first element in our array (NOT the value) 
+# high = the index of the last element in our array (NOT the value) 
+# comparator = a helpermethod which we use to decide how to compare stuff 
+ 
+# Every time we want to sort a picture, we have to pass on the above mentioned 4 parameters. To avoid code duplication, 
+# we created this helpermethod 
+ 
+# This way we only need to manually pass on the array and the comparator method, the helpermethod does the rest for us 
+ 
+def quickSortHelp(arr, comparator): 
+    quickSort(arr, 0, len(arr) - 1, comparator) 
+ 
+ 
+def quickSort(arr, low, high, comparator): 
+    if len(arr) == 1: 
+        return arr 
+    if low < high: 
+        # Pi is the index where the array gets divided in two (the place of our pivot element) 
+        # arr[pi] is sorted 
+        # what partition does will be explained below 
+        pi = partition(arr, low, high, comparator) 
+        # After you've divided the array, you repeat 
+        quickSort(arr, low, pi - 1, comparator) 
+        quickSort(arr, pi + 1, high, comparator) 
+ 
+# Partition actually compares two values with each other 
+def partition(arr, low, high, comparator): 
+    # i : the index of the smallest elements minus 1, so suuuper on the left of the array 
+    i = (low - 1) 
+    # pivot : value of the pivot elements. We get it by using a comparator method. 
+    pivot = comparator(arr[high]) 
+ 
+    for j in range(low, high): 
+        if comparator(arr[j]) <= pivot: 
+            i = i + 1 
+            arr[i], arr[j] = arr[j], arr[i].copy() 
+    arr[i + 1], arr[high] = arr[high], arr[i + 1].copy() 
+    return (i + 1) 
+ 
+ 
+### Comparator = How we determine the value of a pixel 
+# howBright just gives us the sum of the three RGB values 
+def howBright(pixel): 
+    return int(pixel[0]) + int(pixel[1]) + int(pixel[2]) 
+ 
+# Sort by red 
+def valueofRed(pixel): 
+    return int(pixel[0]) 
+ 
+# Oder yellow 
+def valueofYellow(pixel): 
+    return int(pixel[1]) 
+ 
+# Oder blue 
+def valueofBlue(pixel): 
+    return int(pixel[2]) 
+ 
+ 
+ 
+### Convert -> converting each row 
+def convertRow(originalImage): 
```

Comment:
I like the elegance of converting all images to RGBA; but maybe it makes more sense to do that when you load the image in the 1st place so you don't need to worry about that here and don't need to repeat the code for convert column.